src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -315,5 +315,47 @@ public static string[] GetLogicalDrives()
         {
             return FileSystem.GetLogicalDrives();
         }
+
+        /// <summary>
+        /// Creates a directory symbolic link identified by <paramref name="path"/> that points to <paramref name="pathToTarget"/>.
+        /// </summary>
+        /// <param name="path">The absolute path where the symbolic link should be created.</param>
+        /// <param name="pathToTarget">The target directory of the symbolic link.</param>
+        /// <returns>A <see cref="DirectoryInfo"/> instance that wraps the newly created directory symbolic link.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="path"/> or <paramref name="pathToTarget"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="path"/> or <paramref name="pathToTarget"/> is empty.
+        /// -or-
+        /// <paramref name="path"/> is not an absolute path.
+        /// -or-
+        /// <paramref name="path"/> or <paramref name="pathToTarget"/> contains invalid path characters.</exception>
+        /// <exception cref="IOException">A file or directory already exists in the location of <paramref name="path"/>.
+        /// -or-
+        /// An I/O error occurred.</exception>
+        public static FileSystemInfo CreateSymbolicLink(string path, string pathToTarget)
+        {
+            string fullPath = Path.GetFullPath(path);
+            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));
+
+            FileSystem.CreateSymbolicLink(path, pathToTarget, isDirectory: true);
+            return new DirectoryInfo(originalPath: path, fullPath: fullPath, isNormalized: true);
```

Comment:
why are you calling `GetFullPath` before `VerifyValidPath`?

```suggestion
            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));

            FileSystem.CreateSymbolicLink(path, pathToTarget, isDirectory: true);
            return new DirectoryInfo(originalPath: path, fullPath: Path.GetFullPath(path), isNormalized: true);
```

Reply:
To avoid some duplicated checks, see https://github.com/dotnet/runtime/pull/54253#discussion_r661958423.